client/rest-high-level/src/main/java/org/elasticsearch/client/MigrationRequestConverters.java

```diff
@@ -36,4 +38,25 @@ static Request getMigrationAssistance(IndexUpgradeInfoRequest indexUpgradeInfoRe
         parameters.withIndicesOptions(indexUpgradeInfoRequest.indicesOptions());
         return request;
     }
+
+    static Request migrate(IndexUpgradeRequest indexUpgradeRequest) {
+        return prepareMigrateRequest(indexUpgradeRequest, true);
+    }
+
+    static Request submitMigrateTask(IndexUpgradeRequest indexUpgradeRequest) {
+        return prepareMigrateRequest(indexUpgradeRequest, false);
+    }
+
+    private static Request prepareMigrateRequest(IndexUpgradeRequest indexUpgradeRequest, boolean waitForCompletion) {
+        RequestConverters.EndpointBuilder endpointBuilder = new RequestConverters.EndpointBuilder()
+            .addPathPartAsIs("_xpack/migration/upgrade")
```

Comment:
you can comma separate these instead... i know the likelihood of us not using `/` is low, but its best to not have them in this.